framework/src/modules/chain/logic/init_transaction.js

```diff
@@ -101,30 +117,54 @@ const outTransferAsset = raw => {
 	return { outTransfer };
 };
 
+// TODO: remove after https://github.com/LiskHQ/lisk/issues/2424
+const assetDbReadMap = new Map([
+	[0, transferAsset],
+	[1, signatureAsset],
+	[2, delegateAsset],
+	[3, voteAsset],
+	[4, multiAsset],
+	[5, dappAsset],
+	[6, inTransferAsset],
+	[7, outTransferAsset],
+]);
+
 class Transaction {
-	constructor(transactions = {}) {
+	constructor({ registeredTransactions = {} }) {
 		this.transactionClassMap = new Map();
 
-		// TODO: remove after https://github.com/LiskHQ/lisk/issues/2424
-		this.assetDbReadMap = new Map([
-			[0, transferAsset],
-			[1, signatureAsset],
-			[2, delegateAsset],
-			[3, voteAsset],
-			[4, multiAsset],
-			[5, dappAsset],
-			[6, inTransferAsset],
-			[7, outTransferAsset],
-		]);
-
-		Object.keys(transactions).forEach(transactionType => {
+		Object.keys(registeredTransactions).forEach(transactionType => {
 			this.transactionClassMap.set(
 				Number(transactionType),
-				transactions[transactionType]
+				registeredTransactions[transactionType]
 			);
 		});
 	}
 
+	fromBlock(block) {
+		if (!block.transactions) {
+			return undefined;
```

Comment:
shouldnt it return empty array here?

Reply:
it will return empty array if `block.transactions` is an empty array. But if there no `transactions` property I reckon is better to return `undefined`

Reply:
I feel it's safer to return empty array, because if we try to loop through, it wont throw an error